src/app/business/services/instance-list/instance-list.component.html

```diff
@@ -44,11 +44,18 @@
             <button *ngIf="false" [disabled]="instance.status=='requested'" pButton type="button" label="Run" icon="fa fa-play" styleClass="ui-button-success"  (click)="startInstance(instance.id)"></button>
             <button *ngIf="false" [disabled]="instance.status!='requested'" pButton type="button" label="Re-Run" icon="fa fa-repeat" styleClass="ui-button-info"  (click)="restartInstance(instance.id)"></button>
             <button *ngIf="false" pButton type="button" label="Pause" icon="fa fa-pause" styleClass="ui-button-info"  (click)="pauseInstance(instance.id)"></button>
-            <button pButton type="button" label="Delete" icon="fa fa-exclamation-triangle" styleClass="ui-button-danger"  (click)="deleteInstance(instance.id)"></button>
+            <button pButton type="button" label="Delete" icon="fa fa-exclamation-triangle" styleClass="ui-button-danger"  (click)="showWarningDialogFun(instance)"></button>
```

Comment:
I hope we don't need instance.name/instance.id in showWarningDialogFun(instance). Just verify once with multiple instances created and delete one of the instance

Reply:
This is a better experience. User will be able to see the name of the instance and confirm whether they were deleting the correct one.

Reply:
Ok. Actually my question was regarding code. Did you test with multiple instance created? Initially you were passing deleteInstance(instance.id) but now for showing warning you are just using instance object. If it works fine for multiple instances, GTG

Reply:
I am sending the instance object so that I can display the name of the instance before deleting. The first function displays the confirm dialog and then on confirming it invokes the delete instance function where the `id` is passed. This has been tested with multiple instances.